src/cargo/core/resolver/version_prefs.rs

```diff
@@ -39,14 +44,21 @@ impl VersionPreferences {
             .insert(dep);
     }
 
+    pub fn version_ordering(&mut self, ordering: VersionOrdering) {
+        self.version_ordering = ordering;
+    }
+
+    pub fn max_rust_version(&mut self, ver: Option<RustVersion>) {
+        self.max_rust_version = ver;
+    }
+
     /// Sort the given vector of summaries in-place, with all summaries presumed to be for
     /// the same package.  Preferred versions appear first in the result, sorted by
     /// `version_ordering`, followed by non-preferred versions sorted the same way.
     pub fn sort_summaries(
         &self,
         summaries: &mut Vec<Summary>,
-        version_ordering: VersionOrdering,
-        first_version: bool,
+        first_version: Option<VersionOrdering>,
```

Comment:
Can the doc comment be updated? It still refers to `version_ordering`. It might also be nice to add a little bit of text explaining that it can remove summaries, too.